src/DynamoManipulation/NodeManipulator.cs

```diff
@@ -494,19 +494,31 @@ private RenderPackageCache GenerateRenderPackages()
                 return packages;
             }
 
-            // This check is required as for some reason LibG fails to load, geometry nodes are null
-            // and we must return immediately before proceeding with further calls to ProtoGeometry
+            // This check is required if for some reason LibG fails to load, geometry nodes are null
+            // and we must return immediately before proceeding with further calls to ProtoGeometry.
+            // It's possible the the node's return value is not null, but libG was still not loaded correclty.
+            // For example we could be returning a function, and checking for null or function is not sufficent here
+            // as direct manipulators should be created for function outputs. Additionally, checking for Function does not 
+            // indicate that LibG failed to load.
+            // Instead wrap use of libG in a try catch.
             if (IsNodeNull(Node.CachedValue)) return packages;
 
```

Comment:
I think apart from testing for the node to be null, adding the check to see if it's returning a function, and returning if so, will work here, i.e. before even calling `AssignInputNodes`.